### ISSUE TITLE
forall quantification not showing up in constraints and `==` instead of `<=`

### ISSUE THREAD

**Issue by vimuel** [dorchard/granule#18]
```idris
dub : forall n : Nat . Int |n| -> Int
dub |x| = x * x
```

This should not typecheck, but does. 

Relevant debug output:

```
Demand for x1 = _1 : binderk_x1_b1 + _1 : binderk_x1_b1
- Compare for equality Int |_1 : binderk_x1_b1 + _1 : binderk_x1_b1| = Int |n|
_1 : binderk_x1_b1 + _1 : binderk_x1_b1 == n
[ CPlus (COne (CPoly "binderk_x1_b1")) (COne (CPoly "binderk_x1_b1")) , CVar "n"]
+ Compare for equality Int = Int
Ok.
```

**Comment by dorchard** [comment]
Great catch.  This exposed a big flaw in the way the constraints were being generated which I have now fixed with 64f90a9b3aab19126bf7f5cfba3bb79dc934b7e7, 405fcffa6311545ea94c05653f44b601618913ad and e4b2b87d528c1ed853373e93498af5d59828eb09

**Comment by dorchard** [comment]
We now rightly reject this example: https://github.com/dorchard/granule/blob/master/examples/illtyped/universal.gr
